osu.Game/Screens/Play/SongProgressInfo.cs

```diff
@@ -0,0 +1,87 @@
+﻿// Copyright (c) 2007-2017 ppy Pty Ltd <contact@ppy.sh>.
+// Licensed under the MIT Licence - https://raw.githubusercontent.com/ppy/osu/master/LICENCE
+
+using osu.Framework.Allocation;
+using osu.Framework.Graphics;
+using osu.Framework.Graphics.Containers;
+using osu.Framework.Graphics.Primitives;
+using osu.Game.Graphics;
+using osu.Game.Graphics.Sprites;
+using System;
+
+namespace osu.Game.Screens.Play
+{
+    public class SongProgressInfo : Container
+    {
+        private OsuSpriteText timeCurrent;
+        private OsuSpriteText timeLeft;
+        private OsuSpriteText progress;
+
+        private double currentTime;
+        private double songLenght;
+
+        private const int margin = 10;
+
+        public double SongLenght { set { songLenght = value; } }
+        public double CurrentTime
+        {
+            set
+            {
+                currentTime = value;
+                if (value > 0)
+                {
+                    timeCurrent.Text = TimeSpan.FromMilliseconds(value).ToString(@"m\:ss");

```

Comment:
Also calling these `ToString`s every single update is a bit wasteful (`ToString` can get quite expensive). why not check is the current second has changed before doing string formatting?